nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -147,45 +380,62 @@ public NetworkElement registerNetworkElement(List<NetworkElement> portPairGroupL
                                     .portPairGroups(portPairGroupIds)
                                     .build();
 
-        PortChain portChainCreated = this.osClient.sfc().portchains().create(portChain);
+        PortChain portChainCreated = this.osCalls.createPortChain(portChain);
+
+        List<PortPairGroupEntity> portPairGroups =
+                portPairGroupList.stream().map(p -> new PortPairGroupEntity(p.getElementId())).collect(toList());
+
+        ServiceFunctionChainEntity retVal = new ServiceFunctionChainEntity(portChainCreated.getId());
+        portPairGroups.stream().forEach(p -> p.setServiceFunctionChain(retVal));
+        retVal.setPortPairGroups(portPairGroups);
 
-        return new NetworkElementEntity(portChainCreated.getId(),
-                                        new ArrayList<>(), new ArrayList<>(),
-                                        portPairGroupList.get(0).getParentId());
+        return retVal;
     }
 
     @Override
     public NetworkElement updateNetworkElement(NetworkElement serviceFunctionChain, List<NetworkElement> portPairGroupList)
             throws Exception {
-        this.utils.throwExceptionIfNullElementAndId(serviceFunctionChain, "Port Pair Group Service Function Chain Id");
-        this.utils.throwExceptionIfNullOrEmptyNetworkElementList(portPairGroupList, "Port Pair Group update member list");
+        checkArgument(serviceFunctionChain != null && serviceFunctionChain.getElementId() != null,
+                "null passed for %s !", "Service Function Chain Id");
+        throwExceptionIfNullOrEmptyNetworkElementList(portPairGroupList, "Port Pair Group update member list");
 
-        PortChain portChain = this.osClient.sfc().portchains().get(serviceFunctionChain.getElementId());
-        this.utils.throwExceptionIfCannotFindById(portChain, "Service Function Chain", serviceFunctionChain.getElementId());
+        PortChain portChain = this.osCalls.getPortChain(serviceFunctionChain.getElementId());
+        checkArgument(portChain != null && portChain.getId() != null,
+                      "Cannot find %s by id: %s!", "Service Function Chain", serviceFunctionChain.getElementId());
 
         portChain = Builders.portChain().from(portChain)
                             .portPairGroups(Collections.emptyList()).build();
-        this.osClient.sfc().portchains().update(portChain.getId(), portChain);
+        this.osCalls.updatePortChain(portChain.getId(), portChain);
         this.utils.validatePPGList(portPairGroupList);
```

Comment:
do we need this?  SFC API validates it for us already...and would avoid a whole bunch of calls from out side trying to retrive each ppg.

I understand we might have had to do this for for the dummy plugin because the only place validation can happen is in the plugin. But with real SFC has some of the validations already as part of their API and we leverage that as much as possible.